### DISCUSSION TITLE
LOG output between Cooja and hardware is different

### DISCUSSION THREAD

**Discussion by rahav** [contiki-ng/contiki-ng#2702] [Q&A] [answered]
Cooja LOG file contains the `node time` and `id`, in the first and second columns of its output:
> 00:10.236	ID:2	[DBG : Controller] Boot got cycles from OS

Running on hardware the same code will produce
> [DBG : Controller] Boot got cycles from OS

Is there a way or configuration to have the same output format when the code runs on hardware?

**Comment by alexrayne** [comment, marked as answer]
on device you no need ID:? - it always same

printing time too expensive of code and time, so on embedded platform it mostly not fit in. Instead, printing system clock could have some meaning. 

anyway, you can change your log in os/sys/log.c

**Reply by rahav** [reply]
Thank you Alex. By system clock you mean `clock_time()`?

**Reply by rahav** [reply]
Hey @alexrayne and thank you. After thinking about it more, if by "system clock" you meant world-clock or wall-clock, then now it totally makes sense. This way I can interleave the LOG messages from discrete devices.
